Sources/Purchasing/StoreKitAbstractions/StoreProductDiscount.swift

```diff
@@ -70,7 +77,13 @@ public final class StoreProductDiscount: NSObject, StoreProductDiscountType {
     @objc public var offerIdentifier: String? { self.discount.offerIdentifier }
     @objc public var currencyCode: String? { self.discount.currencyCode }
     // See also `priceDecimalNumber` for Objective-C
-    public var price: Decimal { self.discount.price }
+    public var price: Decimal {
+        // See: https://github.com/RevenueCat/purchases-ios/issues/2207
+        // it seems like StoreKit 2 doesn't actually round decimals in a price-friendly fashion, so we do it
+        let priceNSDecimal = (self.discount.price as NSDecimalNumber)
+        return priceNSDecimal.rounding(accordingToBehavior: Self.decimalRoundingBehavior) as Decimal
```

Comment:
for writing tests for this, we could try to reproduce using SKTest. There's a chance that SKTest doesn't actually reproduce, though, and that we end up having to just mock with unit tests.

Which... we can't do directly because SK2 so we'd have to move this logic to some other class or something and test that one instead, then inject it here and test that we're calling the class.

Reply:
here's code that I used to test that the approach works: 

```swift
let price = 114.98999999999997952

print(price)

let priceDecimal = NSDecimalNumber(floatLiteral: price)

print(priceDecimal)

let behavior = NSDecimalNumberHandler(roundingMode: .up, scale: 2, raiseOnExactness: false, raiseOnOverflow: false, raiseOnUnderflow: false, raiseOnDivideByZero: false)

let roundedDecimal = priceDecimal.rounding(accordingToBehavior: behavior)

let swiftDecimal = Decimal(floatLiteral: price)

print(swiftDecimal)

let roundedSwift = (swiftDecimal as NSDecimalNumber).rounding(accordingToBehavior: behavior)

let finalDecimal = roundedSwift as Decimal

print(finalDecimal)

```

Reply:
Based on this I think maybe the best solution is to extract the rounding so we can test it like this, instead of bothering with StoreKit.